lib/ansible/plugins/loader.py

```diff
@@ -113,11 +113,27 @@ def __init__(self, class_name, package, config, subdir, aliases=None, required_b
         if class_name not in PLUGIN_PATH_CACHE:
             PLUGIN_PATH_CACHE[class_name] = defaultdict(dict)
 
+        # hold dirs added at runtime outside of config
+        self._extra_dirs = []
+
+        # caches
         self._module_cache = MODULE_CACHE[class_name]
         self._paths = PATH_CACHE[class_name]
         self._plugin_path_cache = PLUGIN_PATH_CACHE[class_name]
 
-        self._extra_dirs = []
+        self._searched_paths = set()
```

Comment:
It seems like the above 5 lines or so could just be replaced with a call to `self._clear_caches()`

Reply:
i dont want to clear the global caches in init